projects/driver_controls/inc/magnetic_brake_event_generator.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "adc.h"
+#include "ads1015.h"
+#include "ads1015_def.h"
+
+typedef struct MagneticBrakeSettings {
+  int16_t percentage_threshold;  // value above which the brake_pressed event is raised
+  int16_t zero_value;            // lsb value obtained from ads1015 when brake is unpressed
+  int16_t hundred_value;         // lsb value obtained from ads105 when brake is pressed
+  int16_t min_allowed_range;     // min allowed percentage value, usually 0
+  int16_t max_allowed_range;     // max allowed percentage value, (1<<12)
+} MagneticBrakeSettings;
```

Comment:
Remember that our naming scheme dictates that all structs, functions, typedefs, etc. inside a module `foo_bar` should be prefixed `FooBar`, `foo_bar`, etc.

If you were to keep the current name this should be `MagneticBrakeEventGeneratorSettings`.